Mathlib/SetTheory/Game/Birthday.lean

```diff
@@ -157,26 +159,112 @@ theorem birthday_add : ∀ x y : PGame.{u}, (x + y).birthday = x.birthday ♯ y.
     · exact lt_max_of_lt_right ((nadd_le_nadd_left hj _).trans_lt (lt_lsub _ _))
 termination_by a b => (a, b)
 
-theorem birthday_add_zero : (a + 0).birthday = a.birthday := by simp
-
-theorem birthday_zero_add : (0 + a).birthday = a.birthday := by simp
-
-theorem birthday_add_one : (a + 1).birthday = Order.succ a.birthday := by simp
-
-theorem birthday_one_add : (1 + a).birthday = Order.succ a.birthday := by simp
+@[simp]
+theorem birthday_sub (x y : PGame) : (x - y).birthday = x.birthday ♯ y.birthday := by
+  apply (birthday_add x _).trans
+  rw [neg_birthday]
 
 @[simp]
 theorem birthday_natCast : ∀ n : ℕ, birthday n = n
   | 0 => birthday_zero
   | n + 1 => by simp [birthday_natCast]
 
-@[deprecated (since := "2024-04-17")]
-alias birthday_nat_cast := birthday_natCast
+end PGame
 
-theorem birthday_add_nat (n : ℕ) : (a + n).birthday = a.birthday + n := by simp
+namespace Game
 
-theorem birthday_nat_add (n : ℕ) : (↑n + a).birthday = a.birthday + n := by simp
+/-- The birthday of a game is defined as the least birthday among all pre-games that define it. -/
+noncomputable def birthday (x : Game.{u}) : Ordinal.{u} :=
+  sInf (PGame.birthday '' (Quotient.mk' ⁻¹' {x}))
 
-end PGame
+/-- The set in the definition of birthday is nonempty. -/
+theorem birthday_nonempty (x : Game) : (PGame.birthday '' (Quotient.mk' ⁻¹' {x})).Nonempty := by
+  rw [Set.image_nonempty]
+  exact ⟨_, x.out_eq⟩
+
+theorem birthday_eq_pGame_birthday (x : Game) : ∃ y : PGame.{u},
+    ⟦y⟧ = x ∧ y.birthday = birthday x :=
+  csInf_mem (birthday_nonempty x)
+
+theorem birthday_le_pGame_birthday (x : PGame) : birthday ⟦x⟧ ≤ x.birthday :=
+  csInf_le' ⟨x, rfl, rfl⟩
+
+@[simp]
+theorem birthday_zero : birthday 0 = 0 := by
+  rw [← Ordinal.le_zero, ← PGame.birthday_zero]
+  exact birthday_le_pGame_birthday _
+
+@[simp]
+theorem birthday_eq_zero {x : Game} : birthday x = 0 ↔ x = 0 := by
+  constructor
+  · intro h
+    let ⟨y, hy₁, hy₂⟩ := birthday_eq_pGame_birthday x
+    rw [← hy₁]
+    rw [h, PGame.birthday_eq_zero] at hy₂
+    exact PGame.game_eq (@PGame.Equiv.isEmpty _ hy₂.1 hy₂.2)
+  · rintro rfl
+    exact birthday_zero
+
+@[simp]
+theorem toPGame_birthday (o : Ordinal) : birthday o.toGame = o := by
+  apply le_antisymm
+  · conv_rhs => rw [← PGame.toPGame_birthday o]
+    apply birthday_le_pGame_birthday
+  · let ⟨x, hx₁, hx₂⟩ := birthday_eq_pGame_birthday o.toGame
+    rw [← hx₂, ← toPGame_le_iff]
+    rw [← PGame.equiv_iff_game_eq] at hx₁
+    exact hx₁.2.trans (PGame.le_birthday x)
+
+@[simp]
+theorem birthday_natCast (n : ℕ) : birthday n = n := by
```

Comment:
Could you add `birthday_ofNat` too, using `no_index (OfNat.ofNat n)`?

Reply:
What's `no_index`? Does `Game` even have an `OfNat.ofNat` instance? I don't believe we use it anywhere else in it's API.

Reply:
> What's `no_index`?

In practice, a workaround for janky behavior by `simp` around `ofNat`

> Does `Game` even have an `OfNat.ofNat` instance?

Good question; I assumed it would have since it seems to have a `NatCast`, and those usually come together. I didn't actually check though. Is there an `AddMonoidWithOne` instance?

Reply:
Yes, it exists:
```
#synth OfNat SetTheory.Game 37
```

Reply:
Is there any theorem saying `OfNat.ofNat n = ↑n`? There's [Nat.cast_eq_ofNat](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Nat/Cast/Defs.html#Nat.cast_eq_ofNat) but only for naturals greater than 2.

Reply:
No, it should assume `[n.AtLeastTwo]`. The theorem you want is then true by rfl, so you can prove the whole thing in terms of the natCast lemma